In [1]:
from sklearn.datasets import load_iris

The iris object that is returned by load_iris is a Bunch object, which is very similar
 to a dictionary. It contains keys and values:

In [3]:
iris_dataset = load_iris()
print(f"Keys of iris_dataset\n{iris_dataset.keys()}")

Keys of iris_dataset
dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'])


The value of the key DESCR is a short description of the dataset. We show the beginning of the description here (feel free to look up the rest yourself):

In [11]:
print(iris_dataset.DESCR[:193] + "\n...")

.. _iris_dataset:

Iris plants dataset
--------------------

**Data Set Characteristics:**

:Number of Instances: 150 (50 in each of three classes)
:Number of Attributes: 4 numeric, predictive 
...


The value of the key target_names is an array of strings, containing the species of flower that we want to predict:

In [9]:
print(f"Target names: {iris_dataset.target_names}")

Target names: ['setosa' 'versicolor' 'virginica']


 The value of feature_names is a list of strings, giving the description of each feature:

In [13]:
print(f"Feature name: {iris_dataset.feature_names}")

Feature name: ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']


The data itself is contained in the target and data fields. data contains the numeric measurements of sepal length, sepal width, petal length, and petal width in a NumPy array:

In [15]:
print(f"Type of data: {type(iris_dataset.data)}")

Type of data: <class 'numpy.ndarray'>


The rows in the data array correspond to flowers, while the columns represent the four measurements that were taken for each flower:

In [16]:
print(f"Shape of data: {iris_dataset.data.shape}")

Shape of data: (150, 4)


 We see that the array contains measurements for 150 different flowers. Remember that the individual items are called samples in machine learning, and their properties
 are called features.<br> The shape of the data array is the number of samples multiplied by the number of features. This is a convention in scikit-learn, and your data will
 always be assumed to be in this shape.<br> Here are the feature values for the first five samples:

In [18]:
print(f"First five columns:\n{iris_dataset.data[:5]}")

First five columns:
[[5.1 3.5 1.4 0.2]
 [4.9 3.  1.4 0.2]
 [4.7 3.2 1.3 0.2]
 [4.6 3.1 1.5 0.2]
 [5.  3.6 1.4 0.2]]
